jack-test/test/java/com/rapleaf/jack/queries/TestGenericQuery.java

```diff
@@ -926,4 +926,21 @@ public void testIndexHints() throws Exception {
       // expected
     }
   }
+
+  @Test
+  public void testSelectDistinct() throws Exception {
+    String selectStatement = db.createQuery()
+        .from(User.TBL)
+        .select(User.TBL.getAllColumns())
+        .getQueryStatement();
+
+    String selectDistinctStatement = db.createQuery()
+        .from(User.TBL)
+        .selectDistinct(User.TBL.getAllColumns())
```

Comment:
The current syntax support multiple `select` method calls. With the new method, it is possible to write the following code:
```java
.select(col1, col2)
.selectDistinct(col3)
```

This is as if `col1` and `col2` are non-distinct, while `col3` is distinct, which is kind of confusing.

What about a plain `selectDistinct` method that takes no argument? When this method is called, it just flips the `this.selectDistinct` variable:
```java
.select(col1, col2)
.select(col3)
.selectDistinct()
```